modules/aws/src/smithy4s/aws/AwsCredentialsProvider.scala

```diff
@@ -100,7 +100,7 @@ object AwsCredentialsProvider {
   def refreshing[F[_]](
       get: F[AwsTemporaryCredentials]
```

Comment:
Something I didn't consider, we should probably implement some retries because if `get` ever fails we'll stop the loop altogether